File: Latihan_3/python/Mahasiswa.py
class Mahasiswa:
    def __init__(self):
        self.__nama = ""
        self.__nim = ""
    def get_nama(self):
        return self.__nama
    def get_nim(self):
        return self.__nim
    def set_nama(self,nama):
        self.__nama = nama
    def set_nim(self,nim):
        if nim.isdigit() and len(nim) == 10:
            self.__nim = nim
        else:
            print("NIM TIDAK VALID")


mhs = Mahasiswa()
mhs.set_nama("Robby")
mhs.set_nim("1212121212")

print("Nama:", mhs.get_nama())
print("NIM:", mhs.get_nim())
    

File: Latihan_3/Java/Mahasiswa.java
public class Mahasiswa {
    private String nama;
    private String nim;

    public String getNama()
    {
        return nama;
    }
    public String getNim()
    {
        return nim;
    }
    public void setNama(String nama)
    {
        this.nama = nama;
    }

    public void setNim(String nim){
        if(nim != null && nim.matches("\\d{10}")){
            this.nim = nim;
        }
        else{
            System.out.println("NIM tidak valid! Harus 10 digit angka.");
        }
    }

    
    
}
